test/jqLiteSpec.js

```diff
@@ -169,6 +169,75 @@ describe('jqLite', function() {
         expect(nodes[0].nodeName.toLowerCase()).toBe(name);
       });
     });
+
+    describe('security', function() {
+      it('shouldn\'t unsanitize sanitized code', function(done) {
+        // jQuery <3.5.0 fail those tests.
+        if (isJQuery2x()) {
+          done();
+          return;
+        }
+
+        var counter = 0,
+          assertCount = 13,
+          container = jqLite('<div></div>');
+
+        function donePartial() {
+          counter++;
+          if (counter === assertCount) {
+            container.remove();
+            delete window.xss;
+            done();
+          }
+        }
+
+        jqLite(document.body).append(container);
+        window.xss = jasmine.createSpy('xss');
+
+        // Thanks to Masato Kinugawa from Cure53 for providing the following test cases.
+        // Note: below test cases need to invoke the xss function with consecutive
+        // decimal parameters for the assertions to be correct.
+        forEach([
+          '<img alt="<x" title="/><img src=url404 onerror=xss(0)>">',
+          '<img alt="\n<x" title="/>\n<img src=url404 onerror=xss(1)>">',
+          '<style><style/><img src=url404 onerror=xss(2)>',
+          '<xmp><xmp/><img src=url404 onerror=xss(3)>',
+          '<title><title /><img src=url404 onerror=xss(4)>',
+          '<iframe><iframe/><img src=url404 onerror=xss(5)>',
+          '<noframes><noframes/><img src=url404 onerror=xss(6)>',
+          '<noscript><noscript/><img src=url404 onerror=xss(7)>',
+          '<foo" alt="" title="/><img src=url404 onerror=xss(8)>">',
+          '<img alt="<x" title="" src="/><img src=url404 onerror=xss(9)>">',
+          '<noscript/><img src=url404 onerror=xss(10)>',
+          '<noembed><noembed/><img src=url404 onerror=xss(11)>',
+
+          '<option><style></option></select><img src=url404 onerror=xss(12)></style>'
```

Comment:
I think that the following test cases were useful for jQuery (which used a slightly different regex) but not for jqLite's [XHTML_TAG_REGEXP][1] (i.e. they would pass without the changes in this PR, so they don't add much value):

- `'<img alt="<x" title="/><img src=url404 onerror=xss(0)>">'`
- `'<img alt="\n<x" title="/>\n<img src=url404 onerror=xss(1)>">'`
- `'<foo" alt="" title="/><img src=url404 onerror=xss(8)>">'`
- `'<img alt="<x" title="" src="/><img src=url404 onerror=xss(9)>">'`

(That is because jQuery's tegex would match `"` as part of the tag name, while jqLite won't.)

[1]: https://github.com/angular/angular.js/blob/1e45057ae6a7f929a8e37ee7c5ff77f419390b51/src/jqLite.js#L180

Reply:
We’re testing Angular with jQuery as well here. I’d leave them all; the purpose is not to match only what was broken but to prevent future regressions here.